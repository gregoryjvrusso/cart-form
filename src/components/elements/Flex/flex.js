import React from "react";
import { element, string, oneOfType, arrayOf, node } from "prop-types";
import { Container } from "./flex.styles";

const Flex = ({ children, ...props }) => (
  <Container data-testid="flex" {...props}>
    {children}
  </Container>
);

Flex.propTypes = {
  children: oneOfType([element, string, arrayOf(element), node]),
};

Flex.defaultProps = {
  children: undefined,
};

Flex.displayName = "Flex";

export default Flex;
