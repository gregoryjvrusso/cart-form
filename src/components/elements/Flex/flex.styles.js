import styled from "styled-components";
import { flexbox, layout } from "styled-system";

import Box from "../Box";

export const Container = styled(Box)`
  ${{ display: "flex" }}
  ${flexbox}
  ${layout}
`;
