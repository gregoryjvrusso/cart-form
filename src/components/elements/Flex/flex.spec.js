import React from "react";
import { render } from "@testing-library/react";
import Flex from "../Flex";

describe("Flex component", () => {
  it("deve conseguir renderizar [snapshot]", async () => {
    const { container } = render(
      <Flex width={1 / 2} bg="teal" p={10} color="#fff">
        <div></div>
      </Flex>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("verificar se os valores do css estÃ£o corretos", () => {
    const { getByTestId } = render(
      <Flex width={1 / 2} bg="#000" p={10} color="#fff">
        <div></div>
      </Flex>
    );
    expect(getByTestId('flex')).toHaveStyle('width: 50%')
    expect(getByTestId('flex')).toHaveStyle('background-color: #000')
    expect(getByTestId('flex')).toHaveStyle('color: #fff')
  });
});
