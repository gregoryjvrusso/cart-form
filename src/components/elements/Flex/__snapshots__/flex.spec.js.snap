// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex component deve conseguir renderizar [snapshot] 1`] = `
<div
  class="sc-AxjAm hmofGH sc-AxirZ kpHoSj"
  color="#fff"
  data-testid="flex"
  width="0.5"
>
  <div />
</div>
`;
